src/main/java/Duke.java

```diff
@@ -1,10 +1,129 @@
+                handleDone(line);
+            } else if (line.startsWith("todo")) {
+                addTodoTask(line);
+                printTaskConfirmation();
+            } else if (line.startsWith("deadline")) {
+                addDeadlineTask(line);
+                printTaskConfirmation();
+            } else if (line.startsWith("event")) {
+                addEventTask(line);
+                printTaskConfirmation();
+            } else {
+                sayInvalid();
+            }
+        }
+        sayBye();
+    }
+
+    private static void sayHello() {
+        System.out.println(Logo);
+        System.out.println(horizontalLine);
+        System.out.println("Hello! I'm Bobo!");
+        System.out.println("I'm a little blur, but I'd love to help!");
+        System.out.println(horizontalLine);
+    }
+
+    private static void sayBye() {
+        System.out.println(horizontalLine);
+        System.out.println("Ok. Bye bye!");
+        System.out.println(horizontalLine);
+    }
+
+    private static void sayInvalid() {
+        System.out.println(horizontalLine);
+        System.out.println("Sorry I don't understand that! Can you rephrase?");
+        System.out.println(horizontalLine);
+    }
+
+    private static void printList() {
+        System.out.println(horizontalLine);
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < numberOfTasks; i++) {
+            System.out.println((i + 1) + "." + tasks[i]);
+        }
+        System.out.println(horizontalLine);
+    }
+
+    private static void printDone(int taskNumber) {
+        System.out.println(horizontalLine);
+        System.out.println("Okie! Marked this as done:");
+        System.out.println((taskNumber + 1) + "." + tasks[taskNumber]);
+        System.out.println(horizontalLine);
+    }
+
+    private static void handleDone(String line) {
+        int taskNumber = Integer.parseInt(line.substring(END_OF_DONE_INDEX).trim()) - 1;
+        tasks[taskNumber].markAsDone();
+        printDone(taskNumber);
+    }
+
+    private static void printTaskConfirmation() {
+        System.out.println(horizontalLine);
+        System.out.println("Umm ok added:");
+        System.out.println("  " + tasks[numberOfTasks - 1]);
+        System.out.println("Now you have " + numberOfTasks + " tasks in the list.");
+        System.out.println(horizontalLine);
+    }
+
+    private static void addEventTask(String line) {
+        int endOfDescriptionIndex = line.indexOf("/at") - 1;
+        int startOfAtIndex = line.indexOf("/at") + 4;
+        String description = line.substring(END_OF_EVENT_INDEX, endOfDescriptionIndex);
+        String at = line.substring(startOfAtIndex);
+
+        tasks[numberOfTasks] = new Event(description, at);
+        numberOfTasks++;
+    }
+
+    private static void addDeadlineTask(String line) {
+        int endOfDescriptionIndex = line.indexOf("/by") - 1;
+        int startOfByIndex = line.indexOf("/by") + 4;
```

Comment:
Might be better to use a named constant for the number 4, to avoid magic numbers.